syntax = "proto3";
import "google/protobuf/empty.proto";

option csharp_namespace = "DemoApp";

package Shopping;

service ShopKeeper {
  rpc GetItemInfo (ItemInfoRequest) returns (ItemInfoReply);
  rpc GetItemNames (google.protobuf.Empty) returns (stream ItemInfoRequest);
}

message ItemInfoRequest {
  string name = 1;
}

message ItemInfoReply {
	int32 currentStock = 1;
	double unitPrice = 2;
}

